src/main/java/org/broadinstitute/hellbender/tools/walkers/readorientation/LearnReadOrientationModel.java

```diff
@@ -0,0 +1,239 @@
+        return "SUCCESS";
+    }
+
+    @VisibleForTesting
+    public static Histogram<Integer> combineRefHistogramWithRC(final String refContext,
+                                                               final Histogram<Integer> refHistogram,
+                                                               final Histogram<Integer> refHistogramRevComp){
+        Utils.validateArg(refHistogram.getValueLabel()
+                .equals(SequenceUtil.reverseComplement(refHistogramRevComp.getValueLabel())),
+                "ref context = " + refHistogram.getValueLabel() + ", rev comp = " + refHistogramRevComp.getValueLabel());
+        Utils.validateArg(refHistogram.getValueLabel().equals(refContext), "this better match");
+
+        final Histogram<Integer> combinedRefHistogram = F1R2FilterUtils.createRefHistogram(refContext);
+
+        for (final Integer depth : refHistogram.keySet()){
+            final double newCount = refHistogram.get(depth).getValue() + refHistogramRevComp.get(depth).getValue();
+            combinedRefHistogram.increment(depth, newCount);
+        }
+
+        return combinedRefHistogram;
+    }
+
+    @VisibleForTesting
+    public static List<Histogram<Integer>> combineAltDepthOneHistogramWithRC(final List<Histogram<Integer>> altHistograms,
+                                                                             final List<Histogram<Integer>> altHistogramsRevComp){
+        if (altHistograms.isEmpty() && altHistogramsRevComp.isEmpty()){
+            return Collections.emptyList();
+        }
+
+        final String refContext = ! altHistograms.isEmpty() ?
+                F1R2FilterUtils.labelToTriplet(altHistograms.get(0).getValueLabel()).getLeft() :
+                SequenceUtil.reverseComplement(F1R2FilterUtils.labelToTriplet(altHistogramsRevComp.get(0).getValueLabel()).getLeft());
+
+        // Contract: altHistogram must be of the canonical representation of the kmer
+        Utils.validateArg(F1R2FilterConstants.CANONICAL_KMERS.contains(refContext), "refContext must be the canonical representation but got " + refContext);
+
+        final List<Histogram<Integer>> combinedHistograms = new ArrayList<>(F1R2FilterConstants.numAltHistogramsPerContext);
+
+        for (Nucleotide altAllele : F1R2FilterConstants.REGULAR_BASES){
+            // Skip when the alt base is the ref base, which doesn't make sense because this is a histogram of alt sites
+            if (altAllele == F1R2FilterUtils.getMiddleBase(refContext)){
+                continue;
+            }
+
+            final String reverseComplement = SequenceUtil.reverseComplement(refContext);
+            final Nucleotide altAlleleRevComp = Nucleotide.valueOf(SequenceUtil.reverseComplement(altAllele.toString()));
+
+            for (ReadOrientation orientation : ReadOrientation.values()) {
+                final ReadOrientation oppositeType = orientation == ReadOrientation.F1R2 ? ReadOrientation.F2R1 : ReadOrientation.F1R2;
```

Comment:
Can you use `orientation.getOtherOrientation`?

Reply:
yes, done